src/support/mtx_rw.c

```diff
@@ -198,37 +163,75 @@ __wt_readlock_spin(WT_SESSION_IMPL *session, WT_RWLOCK *l)
 void
 __wt_readlock(WT_SESSION_IMPL *session, WT_RWLOCK *l)
```

Comment:
Just noticed: the `__wt_readlock` and `__wt_writelock` functions have read barriers, but the "try" versions don't. I think that's probably a bug.